src/Groups/directproducts.jl

```diff
@@ -0,0 +1,486 @@
+################################################################################
+#
+#  Managing direct products
+#
+################################################################################
+
+export
+    as_matrix_group,
+    as_perm_group,
+    as_polycyclic_group,
+    cartesian_power,
+    direct_product,
+    embedding,
+    factorofdirectproduct,
+    homomorphism_of_semidirect_product,
+    isfull_direct_product,
+    isfull_semidirect_product,
+    isfull_wreath_product,
+    number_of_factors,
+    projection,
+    semidirect_product,
+    sub,
+    wreath_product,
+    write_as_full
+
+
+"""
+    direct_product(L::AbstractVector{<:GAPGroup})
+    direct_product(L::GAPGroup...)
+Return the direct product of the groups in the vector `L`.
+"""
+function direct_product(L::AbstractVector{<:GAPGroup})
+   X = GAP.Globals.DirectProduct(GAP.julia_to_gap([G.X for G in L]))
+   n = length(L)
+   return DirectProductOfGroups(X,L,n,X,true)
+end
+
+function direct_product(L::GAPGroup...)
+   return direct_product([x for x in L])
+end
+
+"""
+    number_of_factors(G::DirectProductOfGroups)
+Return the number of factors of `G`.
+"""
+number_of_factors(G::DirectProductOfGroups) = G.n
+
+"""
+    cartesian_power(G::T, n::Int)
+Return the direct product of `n` copies of `G`, of type `T`.
+"""
+function cartesian_power(G::T, n::Base.Integer) where T <: GAPGroup
+   L = [G for i in 1:n]
+   return direct_product(L)
+end
+
+"""
+    factorofdirectproduct(G::DirectProductOfGroups, j::Int)
+Return the `j`-th factor of `G`.
+"""
+function factorofdirectproduct(G::DirectProductOfGroups, j::Base.Integer)
```

Comment:
This seems to differ from the naming scheme you used above? So how about:
```suggestion
function factor_of_direct_product(G::DirectProductOfGroups, j::Base.Integer)
```

Reply:
Just confused. Fixed it.